demo-transactions/src/main/kotlin/io/holixon/cqrshexagonaldemo/demoparent/transactions/application/CreateAccountUsecase.kt

```diff
@@ -3,17 +3,42 @@ package io.holixon.cqrshexagonaldemo.demoparent.transactions.application
 import io.holixon.cqrshexagonaldemo.demoparent.transactions.application.port.inbound.searchquery.CreateAccountInPort
 import io.holixon.cqrshexagonaldemo.demoparent.transactions.application.port.outbound.account.AccountOutPort
 import io.holixon.cqrshexagonaldemo.demoparent.transactions.application.port.outbound.customer.CustomerOutPort
-import io.holixon.cqrshexagonaldemo.demoparent.transactions.domain.Account
+import io.holixon.cqrshexagonaldemo.demoparent.transactions.domain.event.AccountCreatedEvent
+import io.holixon.cqrshexagonaldemo.demoparent.transactions.domain.model.account.Account
+import io.holixon.cqrshexagonaldemo.demoparent.transactions.domain.model.customer.CustomerNumber
+import io.holixon.cqrshexagonaldemo.demoparent.transactions.domain.service.CustomerAccountVerificationService
+import io.holixon.cqrshexagonaldemo.demoparent.transactions.domain.service.IbanCreationService
+import org.springframework.context.ApplicationEventPublisher
+import org.springframework.stereotype.Component
 
```

Comment:
In the application package, classes should not import framework code.

I would not use `@Component` or the `ApplicationEventPublisher`

For instantiating beans, I would have used the Beans Definition DSL. And I would have written an adapter to publish events